src/adm/settings.rs

```diff
@@ -84,8 +93,31 @@ pub(crate) type AdmSettings = HashMap<String, AdmAdvertiserFilterSettings>;
 /// This allows `CONTILE_ADM_SETTINGS` to either be specified as inline JSON, or if the
 /// Settings are too large to fit into an ENV string, specified in a path to where the
 /// settings more comfortably fit.
-impl From<&Settings> for AdmSettings {
-    fn from(settings: &Settings) -> Self {
+impl From<&mut Settings> for AdmSettings {
+    fn from(settings: &mut Settings) -> Self {
+        // TODO: Convert these to macros.
+        if settings.adm_sub1.is_none() {
+            if settings.sub1.is_some() {
+                settings.adm_sub1 = settings.sub1.clone();
+                warn!(
```

Comment:
Let's switch the `warn!`s to `eprintln!`s -- so they always show up regardless of log levels and more importantly logging isn't initialized yet when this is called, so they'll get lost.